src/HealthGPS/kevin_hall_model.cpp

```diff
@@ -361,6 +504,69 @@ double KevinHallModel::get_weight_quantile(double epa_quantile, core::Gender sex
     return weight_quantiles_.at(sex)[weight_index];
 }
 
+KevinHallAdjustmentTable KevinHallModel::compute_mean_weight(Population &population,
+                                                             std::optional<double> power) const {
+
+    // Local struct to hold count and sum of weight powers.
+    struct SumCount {
+      public:
+        void append(double value) noexcept {
+            sum_ += value;
+            count_++;
+        }
+
+        double mean() const noexcept { return sum_ / count_; }
+
+      private:
+        double sum_{};
+        int count_{};
+    };
+
+    // Compute sums and counts of weight powers for sex and age.
+    auto sumcounts = UnorderedMap2d<core::Gender, int, SumCount>{};
+    sumcounts.emplace_row(core::Gender::female, std::unordered_map<int, SumCount>{});
+    sumcounts.emplace_row(core::Gender::male, std::unordered_map<int, SumCount>{});
+    for (const auto &person : population) {
+        if (!person.is_active()) {
+            continue;
+        }
+
+        double value;
+        if (power.has_value()) {
+            value = pow(person.risk_factors.at("Weight"_id), power.value());
+        } else {
+            value = person.risk_factors.at("Weight"_id);
+        }
+
+        sumcounts.at(person.gender)[person.age].append(value);
+    }
```

Comment:
I like the way you've implemented this, but this code is a perfect example of how things would be easier if HGPS used a `struct`-of-arrays design cf. an array-of-`struct`s one. Imagine if you could get men's and women's weights as two vectors instead. It would make life so much easier!

Reply:
Agree. Soon I hope.